lib/member_page.rb

```diff
@@ -0,0 +1,95 @@
+require 'scraped_page'
+
+class String
+  def tidy
+    gsub(/[[:space:]]+/, ' ').strip
+  end
+end
+
+class MemberPage < ScrapedPage
+  field :url do
+    url
+  end
+
+  field :image do
+    noko.at_css('.dep_img img/@src').text.tidy
+  end
+
+  field :name do
+    name_and_id.split('-')[0..-2]
+               .reject { |w| w == 'ru' }
+               .map(&:upcase)
+               .join(' ')
+  end
+
+  field :name_ru do
+    noko.at_css('.dep_info h1').text.tidy
+  end
+
+  field :id do
+    name_and_id.split('-').pop
+  end
+
+  field :area do
+    area_and_id[0]
+  end
+
+  field :area_id do
+    area_and_id[1]
+  end
+
+  field :telephone do
+    contact_nodes.text
+                 .scan(/\(?\+?[0-9]+-?[0-9]+\)\s[0-9]+-[0-9]+-[0-9]+/)
+                 .map(&:tidy)
+                 .join(';')
+  end
+
+  field :website do
+    contact_nodes.text
+                 .scan(/http:\/\/[a-z]+[.[a-z]+]+/)
+                 .map(&:tidy)
+                 .join(';')
+  end
+
+  field :email do
+    contact_nodes.text
+                 .downcase
+                 .scan(/[a-z]+.?[a-z]+@[a-z]+(?:\.[a-z]+)+/)
+                 .map(&:tidy)
+                 .join(';')
+  end
+
+  field :term do
+    terms[noko.xpath('//span[@class="fp_tab_selected"]').text.to_sym]
```

Comment:
I think this xpath could be usefully moved out to its own method, which might help explain better what's going on here, especially as it's being called from two places.